### PR TITLE
fix(home): small refactoring + some new components and types fixes

### PR DESCRIPTION
**Pull request by helgastogova**
– Added a **use-tailwind-breakpoint** hook for using breakpoints
– Added a type check command into package.json
– Fixed the main page layout (we'll polish it later), added a mobile view for the filters
– Made small refactoring

https://github.com/calcom/examples/assets/3423341/a5cf58bc-c8dc-4077-9f5d-f74ba3b324fd

### CODE REVIEW

**Comment by PeerRich** [comment]
getting a build error 

CleanShot 2024-05-23 at 12 25 54@2x

File: starter/src/app/_components/home/filters.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  Dialog,
+  DialogPortal,
+  DialogOverlay,
+  DialogClose,
+  DialogTrigger,
+  DialogContent,
+} from "@/components/ui/dialog";
+
+export const HomeFilters = () => {
+  const isMd = false; // TODO: add hook to detect screen size
```
Comment (PeerRich) [inline starter/src/app/_components/home/filters.tsx]: this looks cool for tailwind-compatible breakpoints https://github.com/kodingdotninja/use-tailwind-breakpoint

Comment (helgastogova) [reply starter/src/app/_components/home/filters.tsx]: nice!

File: starter/src/app/_components/home/filters.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  Dialog,
+  DialogPortal,
+  DialogOverlay,
+  DialogClose,
+  DialogTrigger,
+  DialogContent,
+} from "@/components/ui/dialog";
+
+export const HomeFilters = () => {
+  const isMd = useBreakpoint("md");
```
Comment (helgastogova) [inline starter/src/app/_components/home/filters.tsx:22]: TODO: isMd needs some time for loading, probably we need a skeleton here while it is loading (later)

File: starter/src/app/_components/home/filters.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    );
+  return (
+    <>
+      <Layout flex fullWidth={false} align="end">
+        <div className="dialog flex flex-row-reverse px-4">
+          <Dialog modal={false}>
+            <DialogTrigger asChild>
+              <Button className="openFilters">Open Filters</Button>
+            </DialogTrigger>
+            <DialogPortal>
```
Comment (helgastogova) [inline starter/src/app/_components/home/filters.tsx:47]: looks a bit ugly, I will fix it later after the atoms

File: starter/src/app/_components/universal/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  flex = false,
+  fullWidth = true,
+  align,
+}: BaseProps & LayoutProps) => {
+  return (
+    <div
+      className={cn(
+        "layout mx-auto my-0",
+        flex && "flex flex-1",
+        align && `justify-${align}`,
```
Comment (p6l-richard) [inline starter/src/app/_components/universal/layout.tsx:35]: Minor: Didn't know template string interpolation worked with tailwind. Could this not work?

File: starter/src/app/_components/universal/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+type LayoutProps = BaseProps & {
+  flex?: boolean;
+  fullWidth?: boolean;
+  align?: "center" | "start" | "end";
+};
+
+export const LayoutAside = ({ children, className }: BaseProps) => {
+  return <aside className={`layout-aside min-w-[260px] p-4 ${className}`}>{children}</aside>;
```
Comment (p6l-richard) [inline starter/src/app/_components/universal/layout.tsx]: Using `cn` here avoids any css specificity issues from manual className interpolation. 

```suggestion
  return <aside className={cn("layout-aside min-w-[260px] p-4", className)`}>{children}</aside>;
```

File: starter/src/app/_components/universal/layout.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const LayoutMain = ({ children, className }: BaseProps) => {
+  return <main className={`layout-main p-4  ${className}`}>{children}</main>;
```
Comment (p6l-richard) [inline starter/src/app/_components/universal/layout.tsx]: Using `cn` here avoids any css specificity issues from manual className interpolation.


```suggestion
  return <main className={cn("layout-main p-4", className)}>{children}</main>;
```

File: starter/src/app/_components/home/filters.tsx (hunk 2 above)
Comment (p6l-richard) [reply starter/src/app/_components/home/filters.tsx:22]: You can avoid the loading state by including both sidebars into the code and apply `hidden md:block` to the version from `md` upwards.

This is how the dashboard navigation works:
https://github.com/calcom/examples/blob/main/starter/src/app/dashboard/layout.tsx

File: starter/src/app/_components/home/filters.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        </Layout.Main>
+      </Layout>
+    );
+  return (
+    <>
+      <Layout flex fullWidth={false} align="end">
+        <div className="dialog flex flex-row-reverse px-4">
+          <Dialog modal={false}>
+            <DialogTrigger asChild>
+              <Button className="openFilters">Open Filters</Button>
```
Comment (p6l-richard) [inline starter/src/app/_components/home/filters.tsx:45]:
```suggestion
            <Button
                variant="outline"
                size="sm"
                className="h-7 gap-1 text-sm"
            >
                <ListFilter className="h-3.5 w-3.5" />
                <span className="sr-only sm:not-sr-only">Filter</span>
            </Button>
```

Comment (p6l-richard) [reply starter/src/app/_components/home/filters.tsx:45]: don't accept as this suggestion isn't properly prettier'd

Just a proposal to use icon & hide the label on mobile

**Review by p6l-richard — APPROVED** [review]
Nice, this is an improvement. Let's ship it!

There are two minor things with the UI that could be addressed. Also left a couple of minor things for className usage, shadcn, breakpoints.

I noticed that the sidebar looks a bit off for responsiveness:
CleanShot 2024-05-23 at 15 52 24

Also, there's seems to be some scrolling issue on mobile:
CleanShot 2024-05-23 at 15 46 26

File: starter/src/app/_components/universal/layout.tsx (hunk 4 above)
Comment (helgastogova) [reply starter/src/app/_components/universal/layout.tsx:35]: you can see that the filters button on the left and the layout centered, so, yes, as I see it works :)

**Comment by helgastogova** [comment]
> Nice, this is an improvement. Let's ship it!
> 
> There are two minor things with the UI that could be addressed. Also left a couple of minor things for className usage, shadcn, breakpoints.
> 
> I noticed that the sidebar looks a bit off for responsiveness: CleanShot 2024-05-23 at 15 52 24 [ CleanShot 2024-05-23 at 15 52 24 ](https://private-user-images.githubusercontent.com/18185649/333216681-5024378e-65f0-4086-9856-ba075b214aa7.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQyNjUsIm5iZiI6MTcxNjQ3Mzk2NSwicGF0aCI6Ii8xODE4NTY0OS8zMzMyMTY2ODEtNTAyNDM3OGUtNjVmMC00MDg2LTk4NTYtYmEwNzViMjE0YWE3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4YmM2NjgwNzdhZTYzYTJhNmQ3MjA3YzI4OTk3YjJiM2FjZGFhZGMzNzYwMTA4MzBlNjhlMTNkMTg3MDc1MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O17-cZdZbfyJgByxNw8BoeQ4L6c6TfW1vNgb91hgxCc) [ ](https://private-user-images.githubusercontent.com/18185649/333216681-5024378e-65f0-4086-9856-ba075b214aa7.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQyNjUsIm5iZiI6MTcxNjQ3Mzk2NSwicGF0aCI6Ii8xODE4NTY0OS8zMzMyMTY2ODEtNTAyNDM3OGUtNjVmMC00MDg2LTk4NTYtYmEwNzViMjE0YWE3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4YmM2NjgwNzdhZTYzYTJhNmQ3MjA3YzI4OTk3YjJiM2FjZGFhZGMzNzYwMTA4MzBlNjhlMTNkMTg3MDc1MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O17-cZdZbfyJgByxNw8BoeQ4L6c6TfW1vNgb91hgxCc)
> 
> Also, there's seems to be some scrolling issue on mobile: CleanShot 2024-05-23 at 15 46 26 [ CleanShot 2024-05-23 at 15 46 26 ](https://private-user-images.githubusercontent.com/18185649/333216776-b5a88f44-f7ea-499f-b7d7-5a3b9b774e1c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQyNjUsIm5iZiI6MTcxNjQ3Mzk2NSwicGF0aCI6Ii8xODE4NTY0OS8zMzMyMTY3NzYtYjVhODhmNDQtZjdlYS00OTlmLWI3ZDctNWEzYjliNzc0ZTFjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MTRjNzY3NDRlMjE2NmVmZWNiNDFiOGQyNWI5MmRiYTc4Yzk0ODc1YzI4ODk4Y2FhNTgyMTFkNGYyMzM3ZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rHShAna95fUR4hzveg8X97-dwSU1INwcSabz5Zvje-4) [ ](https://private-user-images.githubusercontent.com/18185649/333216776-b5a88f44-f7ea-499f-b7d7-5a3b9b774e1c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0NzQyNjUsIm5iZiI6MTcxNjQ3Mzk2NSwicGF0aCI6Ii8xODE4NTY0OS8zMzMyMTY3NzYtYjVhODhmNDQtZjdlYS00OTlmLWI3ZDctNWEzYjliNzc0ZTFjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE0MTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MTRjNzY3NDRlMjE2NmVmZWNiNDFiOGQyNWI5MmRiYTc4Yzk0ODc1YzI4ODk4Y2FhNTgyMTFkNGYyMzM3ZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rHShAna95fUR4hzveg8X97-dwSU1INwcSabz5Zvje-4)

Yes, I just wanted to return tomorrow and make some polish here, in order to merge that PR sooner

File: starter/src/app/_components/home/filters.tsx (hunk 7 above)
Comment (helgastogova) [reply starter/src/app/_components/home/filters.tsx:45]: yes, you are right, that I also left for the polish part

just wanted to merge files changes sooner for not having conflicts

**Pull request merged**